django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1463,7 +1462,7 @@ def get_feast_selector_options(source, folios):
                 folios (list of strs): A list of folios in the source.
 
             Returns:
-                zip object: A zip object combining a list of folios and Feast objects, to be unpacked in template.
+                list of tuples: A list of folios and Feast objects, to be unpacked in template.
```

Comment:
I know this is not you, but I do not like these repeated functions!

Reply:
Yes I don't like them either... There's definitely a way this function can be refactored so that it's not nested within another function and also not repeated throughout the code.